tests/unit/test_json.py

```diff
@@ -81,3 +82,14 @@ def test_deterministic_json_serialization():
     y = {"b": 1, "a": 0}
     data = db_dtypes.JSONArray._from_sequence([y])
     assert data[0] == x
+
+
+def test_to_numpy():
+    data = db_dtypes.JSONArray._from_sequence(JSON_DATA.values())
+    expected = np.asarray(data)
```

Comment:
Are we sure this doesn't call `to_numpy()` behind the scenes? I would be more comfortable if we used `JSON_DATA.values()` or similar (where it's more clear that we're just constructing an array from the kinds of values we expect when unboxing) to get the expected.

```suggestion
    expected = np.asarray([json.dumps(value) for value in JSON_DATA.values()])
```

Reply:
Thanks Tim! Added a new test with more straight data. The `to_numpy()` is copied from `test_json_compliance.py::TestJSONArrayCasting` for running on Python 3.9 environment.